src/main/java/mezz/jei/ingredients/IngredientSorter.java

```diff
@@ -80,4 +103,233 @@ public void invalidateCache() {
+	}
+
+	private Comparator<IIngredientListElementInfo<?>> createAttackComparator() {
+		Comparator<IIngredientListElementInfo<?>> isWeaponComp = 
+			Comparator.comparing(o -> isWeapon(o.getCheatItemStack()));
+		Comparator<IIngredientListElementInfo<?>> attackDamage = 
+			Comparator.comparing(o -> getAttackDamage(o.getCheatItemStack()));
+		Comparator<IIngredientListElementInfo<?>> attackSpeed = 
+			Comparator.comparing(o -> getAttackSpeed(o.getCheatItemStack()));
+		Comparator<IIngredientListElementInfo<?>> maxDamage = 
+			Comparator.comparing(o -> getWeaponDurability(o.getCheatItemStack()));
+		return isWeaponComp.reversed().thenComparing(attackDamage.reversed()).thenComparing(attackSpeed.reversed()).thenComparing(maxDamage.reversed());
+	}
+
+	private Comparator<IIngredientListElementInfo<?>> createArmorComparator() {
+		Comparator<IIngredientListElementInfo<?>> isArmorComp = 
+			Comparator.comparing(o -> isArmor(o.getCheatItemStack()));
+		Comparator<IIngredientListElementInfo<?>> armorSlot = 
+			Comparator.comparing(o -> getArmorSlotIndex(o.getCheatItemStack()));
+		Comparator<IIngredientListElementInfo<?>> armorDamage = 
+			Comparator.comparing(o -> getArmorDamageReduce(o.getCheatItemStack()));
+		Comparator<IIngredientListElementInfo<?>> armorToughness = 
+			Comparator.comparing(o -> getArmorToughness(o.getCheatItemStack()));
+		Comparator<IIngredientListElementInfo<?>> maxDamage = 
+			Comparator.comparing(o -> getArmorDurability(o.getCheatItemStack()));
+		return isArmorComp.reversed().thenComparing(armorSlot.reversed()).thenComparing(armorDamage.reversed()).thenComparing(armorToughness.reversed()).thenComparing(maxDamage.reversed());
+	}
+
+	//ATB:  I mentally don't do well with anonymous functions.
+	private static int getMaxDamage(ItemStack itemStack) {
+		int maxDamage = Integer.MAX_VALUE;
+		if (itemStack != ItemStack.EMPTY) {
+			maxDamage = itemStack.getMaxDamage();	
+		}
+		return maxDamage;
+	};
+
+	private static int getHarvestLevel(ItemStack itemStack) {		
+		if (itemStack != ItemStack.EMPTY) {
+			return (itemStack.getToolTypes().size() > 0 ? itemStack.getHarvestLevel(itemStack.getToolTypes().iterator().next(), null, null) : -1);
+		}
+		return -1;
+	};
+
+	private static boolean isTool(ItemStack itemStack){
+		//Sort non-tools after the tools.
+		return !getToolClass(itemStack).isEmpty();
+	};
+
+	private static int getToolDurability(ItemStack itemStack) {
+		if (isTool(itemStack))
+			return itemStack.getMaxDamage();
+		return 0;
+	}
+
+	private static boolean isWeapon(ItemStack itemStack) {
+		//Sort Weapons apart from tools, armor, and other random things..
+		//0 did not successfully filter out the non-weapons. 1/1024 should do the trick.
+		//AttackDamage also filters out Tools and Armor.  Anything that deals extra damage is a weapon.
+		return getAttackDamage(itemStack) > 0.000976562f;
```

Comment:
`0.000976562f` should be made a constant at the top of the file, and then it can be written as `1.0 / 1024.0` with the explanation there.